external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -210,6 +223,50 @@ pub fn update_compiler_toolchain(
     Ok(())
 }
 
+pub enum ManagedAddressUpdate {
+    Published { original_id: String },
+    Upgraded { latest_id: String, version: u64 },
+}
```

Comment:
This ought to be sufficient to update managed data for `sui client publish` and `sui client upgrade` (in other words, we assume `original_id` is unchanged wrt to an `sui client upgrade`--I'm not sure if we should check that, and if we do, that's probably better off the job of the caller to preserve this invariant)